src/iscsi.pyx

```diff
@@ -89,6 +94,13 @@ cdef class Task:
     def status(self):
         return scsi_task_get_status(self._task, NULL)
 
+    @property
+    def datain(self):
+        return PyBytes_FromStringAndSize(self._task.datain.data, self._task.datain.size)
```

Comment:
I don't think this is the correct function to call for Cython-to-Python return.

This should probably be using the buffer protocol instead: https://cython.readthedocs.io/en/stable/src/userguide/buffer.html

Reply:
Hi @Flameeyes,

I've removed the (unneeded) `datain` property, but have kept the `raw_sense` still implemented using `PyBytes_FromStringAndSize`.

If I understand correctly, the buffer protocol is used when _**sharing**_ memory, but the sense data is `malloc`'d by `libscsi` on **every** `SCSI_STATUS_CHECK_CONDITION`.  (See https://github.com/sahlberg/libiscsi/blob/master/lib/iscsi-command.c#L449).  As such, it seems more sensible to me to allow the `PyBytes_FromStringAndSize` call to allocate a new buffer and copy the data into it.  I'm not sure how the buffer protocol could be made to work with the underlying malloced buffer(s) in the _libiscsi_ code.

A couple of FWIW:
1. I have verified that despite the "String" in the name, `PyBytes_FromStringAndSize` _will_ copy NUL characters.
2. The number of bytes involved is small (typically 18 bytes).

(If you still think the buffer protocol should be used, then I'd greatly appreciate more guidance on how it should be used here.  Thank you.)